apps/apps.humanatlas.io/src/app/pages/us1/cell-population-predictions/cell-population-predictions.component.ts

```diff
@@ -98,4 +113,28 @@ export class CellPopulationPredictionsComponent {
       this.router.navigate(['/us1']);
     }
   }
+
+  /** Triggered when clicked on download CSV button  */
+  onDownloadCSVButtonClicked() {
+    const fields = this.displayedColumns as (keyof CellSummaryRow)[];
+    const headers = Object.values(this.columnHeaders).join(',') + '\n';
+    const data = this.dataSource.data;
+    const rows = data
+      .map((row) =>
+        fields
+          .map((field) => {
+            if (field === 'percentage') {
+              return row.percentage < 0.0001 ? '< 0.01%' : (row.percentage * 100).toFixed(2) + '%';
+            }
+            return row[field];
+          })
+          .join(','),
+      )
+      .join('\n');
+
+    const csvString = headers + rows;
```

Comment:
Use papaparse to unparse the data into csv. It will handle escaping and other edge cases for us